src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,122 @@
+package duke;
+
+import duke.command.*;
+import duke.exception.IllegalCommandException;
+import duke.exception.IllegalDeleteArgument;
+import duke.exception.IllegalDoneArgument;
+import duke.exception.NoDescriptionException;
+import duke.exception.NoTimeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * Class to understand user's command.
+ * The essential functionality of this helper class is to map user command to
+ * <code>Command</code> objects.
+ */
+public class Parser {
+
+    /**
+     * Map a response string to an executable <code>Command</code> object.
+     * This methods map a response string to an executable <code>Command</code>
+     * object. When converting, commands that Duke cannot understand will lead
+     * to exception throwing.
+     * @param response one-line input from the user
+     * @return a <code>Command</code> object that Duke can execute
+     * @throws Exception exceptions when Duke cannot understands the command or
+     *         when the command does not have the required format.
+     */
+    public Command parse(String response) throws Exception {
+        switch (response) {
```

Comment:
I think it is better to merge the 2 switch cases. Furthermore, your switch cases here should have a default case.